fairing/ml_tasks/tasks.py

```diff
@@ -0,0 +1,62 @@
+import logging
+
+import fairing
+from fairing.deployers.job.job import Job
+from fairing.deployers.serving.serving import Serving
+from .utils import guess_preprocessor, guess_builder
+
+logger = logging.getLogger(__name__)
+
+
+class BaseTask:
+    """
+    Base class for handling high level ML tasks.
+    args:
+        entry_point: An object or reference to the source code that has to be deployed.
+        base_docker_image: Name of the base docker image that should be used as a base image
+            when building a new docker image as part of an ML task deployment.
+        docker_registry: Docker registry to store output docker images.
+        input_files: list of files that needs to be packaged along with the entry point.
+            E.g. local python modules, trained model weigths, etc.
+    """
+    def __init__(self, entry_point, base_docker_image, docker_registry, input_files=[]):
+        preprocessor = guess_preprocessor(entry_point, input_files=input_files)
+        logger.warn("Using preprocessor: {}".format(type(preprocessor)))
+
+        self.docker_registry = docker_registry
+        logger.warn("Using docker registry: {}".format(self.docker_registry))
+
+        gussed_builder = guess_builder(needs_deps_installation=True)
+        logger.warn("Using builder: {}".format(gussed_builder.__name__))
+
+        self.builder = gussed_builder(preprocessor=preprocessor,
+                                      base_image=base_docker_image,
+                                      registry=self.docker_registry)
+
+    def _build(self):
+        self.builder.build()
+        self.pod_spec = self.builder.generate_pod_spec()
```

Comment:
How will pod_spec_mutators be handled in the high level API?

Reply:
I am thinking about how to enable more granular configuration like pod_spec_mutators when users are using these high level APIs. Couple of approach I am thinking: users can use fairing.config.set_<>(..) to set any granular configuration or a backend class (Kubeflow, GCP, etc) can set these good defaults or combination of both. But to keep this PR manageable, let's address that in future PRs. This topic needs more discussion.